### PR TITLE
CDN Redirection

### PR DESCRIPTION
**Pull request by cristinamanum**
App Service to redirect the api.nuget.org calls to the China CDN. The majority of the code is auto generated. The redirection is done in Web.config. Status/Index is used for checking the app status.

### CODE REVIEW

File: src/NuGetCDNRedirect/Content/Site.css [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿body {
```
Comment (joelverhagen) [inline src/NuGetCDNRedirect/Content/Site.css]: Is this site user facing? If not, why have any CSS or .cshtml at all?

File: src/NuGetCDNRedirect/App_Start/RouteConfig.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+using System.Web.Routing;
+
+namespace NuGetCDNRedirect
```
Comment (joelverhagen) [inline src/NuGetCDNRedirect/App_Start/RouteConfig.cs]: Namespace should be `NuGet.Services.<something>`

File: src/NuGetCDNRedirect/Web.config [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    </assemblyBinding>
+  </runtime>
+  <system.webServer>
+    <validation validateIntegratedModeConfiguration="false" />
+    <modules>
+      <remove name="ApplicationInsightsWebTracking" />
+      <add name="ApplicationInsightsWebTracking" type="Microsoft.ApplicationInsights.Web.ApplicationInsightsHttpModule, Microsoft.AI.Web" preCondition="managedHandler" />
+    </modules>
+    <httpRedirect enabled="true" httpResponseStatus="Found">
+      <add wildcard="*" destination="#{Application.CDNRedirect}"/>
```
Comment (joelverhagen) [inline src/NuGetCDNRedirect/Web.config]: What is this `#{Application.CDNRedirect` placeholder thing? Where does the destination value come from?

Comment (cristinamanum) [inline src/NuGetCDNRedirect/Web.config]: Octopus. Octopus project name is: CDN Redirection

File: src/NuGetCDNRedirect/NuGetCDNRedirect.csproj [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    <Content Include="favicon.ico" />
+    <Content Include="fonts\glyphicons-halflings-regular.svg" />
+    <Content Include="Global.asax" />
+    <Content Include="Content\Site.css" />
+    <Content Include="ApplicationInsights.config">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <None Include="Properties\PublishProfiles\NuGetCDNDev1CN - Web Deploy.pubxml" />
+    <None Include="Properties\PublishProfiles\NuGetCDNDevCN - Web Deploy.pubxml" />
+    <None Include="Properties\PublishProfiles\NuGetCDNRedirectDev - Web Deploy.pubxml" />
```
Comment (shishirx34) [inline src/NuGetCDNRedirect/NuGetCDNRedirect.csproj]: are these all profiles needed?

File: src/NuGetCDNRedirect/Content/Site.css (hunk 1 above)
Comment (cristinamanum) [inline src/NuGetCDNRedirect/Content/Site.css]: It is not user facing. Updated.

File: src/NuGetCDNRedirect/App_Start/RouteConfig.cs (hunk 2 above)
Comment (cristinamanum) [inline src/NuGetCDNRedirect/App_Start/RouteConfig.cs]: Updated.

File: src/NuGetCDNRedirect/NuGetCDNRedirect.csproj (hunk 4 above)
Comment (cristinamanum) [inline src/NuGetCDNRedirect/NuGetCDNRedirect.csproj]: Used for local deployment. Removed.

**Comment by skofman1** [comment]
> using System;

file header

---
Refers to: src/NuGetCDNRedirect/Global.asax.cs:1 in f46176b. [](commit_id = f46176b89bd3514b53a0a2e1bc8272309d225c5c, deletion_comment = False)

File: src/NuGetCDNRedirect/NuGetCDNRedirect.nuspec [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    <projectUrl>https://github.com/NuGet/NuGet.Jobs</projectUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>NuGetCDNRedirect</description>
+    <tags>nuget job</tags>
+  </metadata>
+  <files>
+    	<file src="obj\$configuration$\Package\PackageTmp\*.*" />
+    	<file src="obj\$configuration$\Package\PackageTmp\bin\*.*" target="bin"/>
+	<file src="obj\$configuration$\Package\PackageTmp\bin\roslyn\*.*" target="bin\roslyn"/>
+	<file src="obj\$configuration$\Package\PackageTmp\Content\*.*" target="Content"/>
```
Comment (skofman1) [inline src/NuGetCDNRedirect/NuGetCDNRedirect.nuspec]: some of those can be cleaned. like Content

File: src/NuGetCDNRedirect/Controllers/HomeController.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿
+using System.Web.Mvc;
+
+namespace NuGet.Services.CDNRedirect.Controllers
+{
+    public class HomeController : Controller
+    {
+        public ActionResult Index()
+        {
+            return new HttpNotFoundResult("Resource not found.");
```
Comment (skofman1) [inline src/NuGetCDNRedirect/Controllers/HomeController.cs]: perhaps provide details about the service here instead of 404? Maybe "Redirection service is available. Redirection endpoint is .."

**Comment by skofman1** [comment]
:shipit:

File: src/NuGetCDNRedirect/packages.config [hunk 7]
```diff
@@ -0,0 +1,26 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Antlr" version="3.4.1.9004" targetFramework="net461" />
+  <package id="bootstrap" version="3.0.0" targetFramework="net461" />
+  <package id="jQuery" version="1.10.2" targetFramework="net461" />
```
Comment (joelverhagen) [inline src/NuGetCDNRedirect/packages.config]: We don't need `jQuery`, `bootstrap`, `Modernizr`, and others

File: src/NuGetCDNRedirect/Controllers/StatusController.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿
+using System.Web.Mvc;
+
+namespace NuGet.Services.CDNRedirect.Controllers
+{
+    public class StatusController : Controller
```
Comment (joelverhagen) [inline src/NuGetCDNRedirect/Controllers/StatusController.cs]: Do we really need another controller for this?

File: src/NuGetCDNRedirect/Controllers/HomeController.cs (hunk 6 above)
Comment (cristinamanum) [inline src/NuGetCDNRedirect/Controllers/HomeController.cs]: Updated - I let only the status and deleted the above.

File: src/NuGetCDNRedirect/packages.config (hunk 7 above)
Comment (cristinamanum) [inline src/NuGetCDNRedirect/packages.config]: Removed a set of packages that did not seem used.

File: src/NuGetCDNRedirect/Controllers/StatusController.cs (hunk 8 above)
Comment (cristinamanum) [inline src/NuGetCDNRedirect/Controllers/StatusController.cs]: I need a way of checking the app health. The controller is basic.

File: src/NuGetCDNRedirect/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+//
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Revision and Build Numbers
+// by using the '*' as shown below:
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (chenriksson) [inline src/NuGetCDNRedirect/Properties/AssemblyInfo.cs]: Should we set version from build script like elsewhere? Also, the Guid for COM shouldn't be necessary.

File: src/NuGetCDNRedirect/Properties/AssemblyInfo.cs [hunk 10]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Reflection;
```
Comment (chenriksson) [inline src/NuGetCDNRedirect/Properties/AssemblyInfo.cs]: Need copyright header

File: build.ps1 [hunk 11]
```diff
@@ -53,6 +53,22 @@ Function Prepare-Vcs-Callback {
     Build-Solution $Configuration $BuildNumber -MSBuildVersion "$msBuildVersion" "src\Validation.Callback.Vcs\Validation.Callback.Vcs.csproj" -Target "Package" -MSBuildProperties "/P:PackageLocation=obj\Validation.Callback.Vcs.zip" -SkipRestore
 }
 
+Function Prepare-NuGetCDNRedirect {
+    [CmdletBinding()]
```
Comment (chenriksson) [inline build.ps1]: Can Prepare-Vcs-Callback and Prepare=NuGetCDNRedirect be combined into a more general Prepare-NuGetPackage routine?

File: NuGet.Jobs.sln [hunk 12]
```diff
@@ -85,6 +85,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Validation.Helper", "src\Va
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Validation.Helper.Tests", "tests\Validation.Helper.Tests\Validation.Helper.Tests.csproj", "{8336BEBE-EC6A-4E40-A1C9-8C34A507E62D}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "NuGetCDNRedirect", "src\NuGetCDNRedirect\NuGetCDNRedirect.csproj", "{BC9EA7CE-AD21-4D17-B581-F8ED8CBD7191}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
```
Comment (chenriksson) [inline NuGet.Jobs.sln]: I kind of wonder if the Gallery repo is a better fit for this CDN redirect site.

Comment (chenriksson) [inline NuGet.Jobs.sln]: Would also be nice to have common targets file for consistency across sites; i.e. for common config, to enable CodeAnalysis/TreatWarningsAsErrors, etc. Maybe a common AppInsights.config as well.

File: src/NuGetCDNRedirect/Properties/AssemblyInfo.cs (hunk 10 above)
Comment (cristinamanum) [inline src/NuGetCDNRedirect/Properties/AssemblyInfo.cs]: Updated

File: src/NuGetCDNRedirect/Properties/AssemblyInfo.cs (hunk 9 above)
Comment (cristinamanum) [inline src/NuGetCDNRedirect/Properties/AssemblyInfo.cs]: Updated

File: build.ps1 (hunk 11 above)
Comment (cristinamanum) [inline build.ps1]: I prefer to let them independent.

File: NuGet.Jobs.sln (hunk 12 above)
Comment (cristinamanum) [inline NuGet.Jobs.sln]: This service is not related with gallery.
